### PR TITLE
Debugging in VS Code Cookbook Recipe

### PR DESCRIPTION
**Pull request by sdras**

### CODE REVIEW

**Review by NataliaTepluhina — APPROVED** [review]
Thank you Sarah, this looks perfectly good to me! 🎉

File: src/cookbook/debugging-in-vscode.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+Before you can debug your Vue components from VS Code, you need to update the generated Webpack config to build sourcemaps. We do this so that our debugger has a way to map the code within a compressed file back to its position in the original file. This ensures that you can debug an application even after your assets have been optimized by Webpack.
+
+If you use Vue CLI 2, set or update the `devtool` property inside `config/index.js`:
+
+```json
+devtool: 'source-map',
+```
+
+If you use Vue CLI 3, set or update the `devtool` property inside `vue.config.js`:
```
Comment (skirtles-code) [inline src/cookbook/debugging-in-vscode.md:26]: Should this be updated to Vue CLI 4?

Is mentioning CLI 2 still worthwhile in a Vue 3 context?

File: src/cookbook/debugging-in-vscode.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  configureWebpack: {
+    devtool: 'source-map',
+  },
+}
+```
+
+### Launching the Application from VS Code
+
+::: info
+We're assuming the port to be `8080` here. If it's not the case (for instance, if `8080` has been taken and Vue CLI automatically picks another port for you), just modify the configuration accordingly.
```
Comment (skirtles-code) [inline src/cookbook/debugging-in-vscode.md:39]: I stumbled a bit on this. Specifically the *'If it's not the case'* part.

To my ear, it should be either *'If that's not the case'* or just *'If it's not'*.

File: src/cookbook/debugging-in-vscode.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      "webRoot": "${workspaceFolder}/src",
+      "pathMappings": [{ "url": "webpack:///src/", "path": "${webRoot}/" }]
+    }
+  ]
+}
+```
+
+## Setting a Breakpoint
+
+1.  Set a breakpoint in **src/components/HelloWorld.vue** on `line 90` where the `data` function returns a string.
```
Comment (skirtles-code) [inline src/cookbook/debugging-in-vscode.md:75]: Ultra pedantic, but the function is returning an object, not a string.

File: src/cookbook/debugging-in-vscode.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+4.  Your breakpoint should now be hit as a new browser instance opens `http://localhost:8080`.
+
+![Breakpoint Hit](/images/breakpoint_hit.png)
+
+## Alternative Patterns
+
+### Vue Devtools
+
+There are other methods of debugging, varying in complexity. The most popular and simple of which is to use the excellent Vue.js devtools [for Chrome](https://chrome.google.com/webstore/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd) and [for Firefox](https://addons.mozilla.org/en-US/firefox/addon/vue-js-devtools/). Some of the benefits of working with the devtools are that they enable you to live-edit data properties and see the changes reflected immediately. The other major benefit is the ability to do time travel debugging for Vuex.
```
Comment (skirtles-code) [inline src/cookbook/debugging-in-vscode.md:95]: These seem to be links to the Vue 2 devtools. Do they need changing to the Vue 3 devtools beta?

File: src/cookbook/debugging-in-vscode.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+![Devtools Timetravel Debugger](/images/devtools-timetravel.gif)
+
+Please note that if the page uses a production/minified build of Vue.js (such as the standard link from a CDN), devtools inspection is disabled by default so the Vue pane won't show up. If you switch to an unminified version, you may have to give the page a hard refresh to see them.
+
+### Simple Debugger Statement
+
+The example above has a great workflow. However, there is an alternative option where you can use the [native debugger statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger) directly in your code. If you choose to work this way, it's important that you remember to remove the statements when you're done.
+
+```js
```
Comment (skirtles-code) [inline src/cookbook/debugging-in-vscode.md:105]: The language doesn't currently match the code. Either the language needs changing or the `<script>` wrapper should be removed.

**Review by skirtles-code — COMMENTED** [review]
I know this has already been merged but here are some thoughts from me.

**Comment by DannyFeliz** [comment]
@skirtles-code can you make a PR with your suggestions? so we don't lose them.

**Pull request merged**